src/masternodes/mn_checks.cpp

```diff
@@ -3083,6 +3097,93 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
+        if (!attributes->GetValue(activeKey, false))
+            return Res::Err("Payback of DUSD loan with collateral is not currently active");
+
+        // vault exists
+        auto vault = mnview.GetVault(obj.vaultId);
+        if (!vault)
+            return Res::Err("Vault <%s> not found", obj.vaultId.GetHex());
+
+        // vault under liquidation
+        if (vault->isUnderLiquidation)
+            return Res::Err("Cannot payback vault with collateral while vault's under liquidation");
+
+        // owner auth
+        if (!HasAuth(vault->ownerAddress))
+            return Res::Err("tx must have at least one input from token owner");
+
+        if (!IsVaultPriceValid(mnview, obj.vaultId, height))
+            return Res::Err("Cannot payback vault with collateral while any of the asset's price is invalid");
+
+        const auto collateralAmounts = mnview.GetVaultCollaterals(obj.vaultId);
+        if (!collateralAmounts) {
+            return Res::Err("Vault has no collaterals");
+        }
+        if (!collateralAmounts->balances.count(dUsdToken->first)) {
+            return Res::Err("Vault does not have any DUSD collaterals");
+        }
+        const auto& collateralDUSD = collateralAmounts->balances.at(dUsdToken->first);
+
+        const auto loanAmounts = mnview.GetLoanTokens(obj.vaultId);
+        if (!loanAmounts) {
+            return Res::Err("Vault has no loans");
+        }
+        if (!loanAmounts->balances.count(dUsdToken->first)) {
+            return Res::Err("Vault does not have any DUSD loans");
+        }
+        const auto& loanDUSD = loanAmounts->balances.at(dUsdToken->first);
+
+        auto rate = mnview.GetInterestRate(obj.vaultId, dUsdToken->first, height);
+        if (!rate)
+            return Res::Err("Cannot get interest rate for this token (DUSD)!");
+        auto subInterest = TotalInterest(*rate, height);
+
+        // Edge case where interest is greater than collateral
+        if (subInterest > collateralDUSD) {
+            res = mnview.SubVaultCollateral(obj.vaultId, {dUsdToken->first, collateralDUSD});
+            if (!res)
+                return res;
+
+            return mnview.DecreaseInterest(height, obj.vaultId, vault->schemeId, dUsdToken->first, 0, collateralDUSD);
+        }
+
+        CTokenAmount subLoanAmount;
+        CTokenAmount subCollateralAmount;
+        if (loanDUSD + subInterest > collateralDUSD) {
+            subLoanAmount = {dUsdToken->first, collateralDUSD - subInterest};
+            subCollateralAmount = {dUsdToken->first, collateralDUSD};
+        } else {
+            subLoanAmount = {dUsdToken->first, loanDUSD};
+            subCollateralAmount = {dUsdToken->first, loanDUSD + subInterest};
+        }
+
+        res = mnview.SubLoanToken(obj.vaultId, subLoanAmount);
+        if (!res)
+            return res;
+
+        res = mnview.SubVaultCollateral(obj.vaultId, subCollateralAmount);
+        if (!res)
+            return res;
+
```

Comment:
missing the sub from minted amount. subCollateralAmount is taken out of circulation -> needs to reduce the total minted amount.

Reply:
Fixed in https://github.com/DeFiCh/ain/pull/1451/commits/99161a1ce7889ace1d42f260d4fc90c64774baff